driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -112,20 +118,33 @@ public InternalStreamConnection(final ServerId serverId, final StreamFactory str
         this.commandListener = commandListener;
         this.connectionInitializer = notNull("connectionInitializer", connectionInitializer);
         description = new ConnectionDescription(serverId);
+        initialServerDescription = ServerDescription.builder()
+                .address(serverId.getAddress())
+                .type(ServerType.UNKNOWN)
+                .state(ServerConnectionState.CONNECTING)
+                .build();
     }
 
     @Override
     public ConnectionDescription getDescription() {
         return description;
     }
 
+    @Override
+    public ServerDescription getInitialServerDescription() {
+       return initialServerDescription;
+    }
+
     @Override
     public void open() {
         isTrue("Open already called", stream == null);
         stream = streamFactory.create(serverId.getAddress());
         try {
             stream.open();
-            description = connectionInitializer.initialize(this);
+            LOGGER.debug("Done opening stream to " + serverId.toString());
```

Comment:
Is this a debugging log statement?  

nit: `if (LOGGER.isDebugEnabled()) {`

Reply:
Removed